common/common-rest/src/main/java/io/servicecomb/common/rest/filter/HttpClientFilter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Huawei Technologies Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.common.rest.filter;
+
+import io.servicecomb.core.Invocation;
+import io.servicecomb.swagger.invocation.Response;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+
+public interface HttpClientFilter {
+  int getOrder();
+
+  // bodyBytes maybe be null
+  void beforeSendRequest(Invocation invocation, HttpClientRequest clientRequest, Buffer requestBodyBuffer);
+
+  // if finished, then return a none null response
```

Comment:
It's better to give a java doc comment.  spell  non null ? or a response that is not null